### PR TITLE
Added WaterCommandFactory and ViewWaterCommand classes

### LINKED ISSUE #115: Create Command to view water

### LINKED ISSUE #121: Implement Parser support to add water intake

### LINKED ISSUE #122: Implement Parser support to view water intake

### LINKED ISSUE #127: Implement Parser support to delete water intake

### PR DESCRIPTION
**Pull request by Atulteja**
Close #127
Close #122 
Close #121 
Close #115

### CODE REVIEW

File: src/main/java/command/water/ViewWaterCommand.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    public static final String COMMAND_WORD = "view";
+
+    protected final LocalDate date;
+
+    public ViewWaterCommand(LocalDate date) {
+        this.date = date;
+    }
+
+    public CommandResult execute(History history) {
+        Water waterList = getWaterList(history);
```
Comment (Bev-low) [inline src/main/java/command/water/ViewWaterCommand.java:19]: actually, for this, can do this way
DailyRecord dailyRecord = history.getRecordByDate(date);
        MealList meals = dailyRecord.getMealList();
        return new CommandResult(meals.toString());
        
The way we access water and mealList is through DailyRecord

File: src/main/java/parser/command/WaterCommandFactory.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        case AddWaterCommand.COMMAND_WORD -> prepareAddCommand(arguments);
+        case DeleteWaterCommand.COMMAND_WORD -> prepareDeleteCommand(arguments);
+        case ViewWaterCommand.COMMAND_WORD -> prepareViewCommand(arguments);
+        default -> new InvalidCommand();
+        };
+    }
+
+    public Command prepareAddCommand(String argumentString) {
+        FlagParser flagParser = new FlagParser(argumentString);
+        flagParser.validateRequiredFlags("/d");
```
Comment (Bev-low) [inline src/main/java/parser/command/WaterCommandFactory.java:34]: these validateRequiredFlags should not be /d as /d stands for days

File: src/main/java/parser/command/WaterCommandFactory.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+        float water = flagParser.getFloatByFlag("w");
+        LocalDate date = flagParser.getDateByFlag("/t");
+
+        return new AddWaterCommand(water, date);
+    }
+
+    public Command prepareDeleteCommand(String argumentString) {
+        FlagParser flagParser = new FlagParser(argumentString);
+        flagParser.validateRequiredFlags("/d");
```
Comment (Bev-low) [inline src/main/java/parser/command/WaterCommandFactory.java:44]: same for here

File: src/main/java/parser/command/WaterCommandFactory.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+        int waterIndexToDelete = flagParser.getIntegerByFlag("w");
+        LocalDate date = flagParser.getDateByFlag("/t");
+
+        return new DeleteWaterCommand(waterIndexToDelete, date);
+    }
+
+    public Command prepareViewCommand(String argumentString) {
+        FlagParser flagParser = new FlagParser(argumentString);
+        flagParser.validateRequiredFlags("/d");
```
Comment (Bev-low) [inline src/main/java/parser/command/WaterCommandFactory.java:54]: and here should not have a validate

**Review by Bev-low — CHANGES_REQUESTED** [review]
Overall is okay but some changes are needed

File: src/main/java/command/water/ViewWaterCommand.java (hunk 1 above)
Comment (Atulteja) [reply src/main/java/command/water/ViewWaterCommand.java:19]: it already does that in the getWaterList fucntion, same thing for meal as well, the naming is different though

**Review by Bev-low — APPROVED** [review]

File: src/main/java/parser/ParserUtils.java [hunk 5]
```diff
@@ -33,6 +33,21 @@ public static int parseInteger(String intString){
         }
     }
 
+    public static float parseFloat(String floatString) {
+        assert floatString != null : "floatString must not be null";
+
+        String trimmedFloatString = floatString.trim();
```
Comment (nirala-ts) [inline src/main/java/parser/ParserUtils.java:39]: LGTM

**Review by nirala-ts — APPROVED** [review]

**Pull request merged**